echopype/calibrate/cal_params.py

```diff
@@ -1,155 +1,249 @@
+            )
 
 
-def get_cal_params_EK(
-    beam: xr.Dataset, vend: xr.Dataset, user_cal_dict: Dict[str, xr.DataArray]
-) -> Dict:
+def sanitize_user_cal_dict(
+    sonar_type: str,
+    user_dict: Dict[str, Union[int, float, xr.DataArray]],
+    channel: Union[List, xr.DataArray],
+) -> Dict[str, Union[int, float, xr.DataArray]]:
     """
-    Get cal params using user inputs or values from data file.
+    Create a blue print for cal_params dict and check the format/organize user-provided params.
 
     Parameters
     ----------
-    beam : xr.Dataset
-        A subset of Sonar/Beam_groupX that contains only the channels specified for calibration
-    vend : xr.Dataset
-        A subset of Vendor_specific that contains only the channels specified for calibration
-    user_cal_dict : dict
-        A dictionary containing user-defined calibration parameters.
-        The user-defined calibration parameters will overwrite values in the data file.
-
-    Returns
-    -------
-    A dictionary containing the calibration parameters for the EK echosounders
+    sonar_type : str
+        type of sonar, one of "EK60", "EK80", or "AZFP"
+    user_dict : dict
+        a dict containing user input calibration parameters as {parameter name: parameter value}
+        parameter value has to be a scalar (int or float) or an xr.DataArray
+        if parameter value is an xr.DataArray, it has to either have 'channel' as a coorindate
+        or have both 'cal_channel_id' and 'cal_frequency' as coordinates
+
+    channel : list or xr.DataArray
+        a list of channels to be calibrated
+        for EK80 data, this list has to corresponds with the subset of channels
+        selected based on waveform_mode and encode_mode
     """
-    out_dict = dict.fromkeys(CAL_PARAMS["EK"])
-    if user_cal_dict is None:
-        user_cal_dict = {}
+    # Check sonar type
+    if sonar_type not in ["EK60", "EK80", "AZFP"]:
+        raise ValueError("'sonar_type' has to be one of: 'EK60', 'EK80', or 'AZFP'")
 
-    # Params from the Beam group
-    params_from_beam = [
-        "angle_offset_alongship",
-        "angle_offset_athwartship",
-        "beamwidth_twoway_alongship",
-        "beamwidth_twoway_athwartship",
-        "equivalent_beam_angle",
-    ]
-    for p in params_from_beam:
-        # substitute if p not in user input
-        out_dict[p] = user_cal_dict.get(p, beam[p])
-
-    # Params from the Vendor_specific group
-    params_from_vend = ["sa_correction", "gain_correction"]
-    for p in params_from_vend:
-        # substitute if p not in user input
-        out_dict[p] = user_cal_dict.get(p, get_vend_cal_params_power(beam=beam, vend=vend, param=p))
+    # Make channel a sorted list
+    if not isinstance(channel, (list, xr.DataArray)):
+        raise ValueError("'channel' has to be a list or an xr.DataArray")
+    else:
+        if isinstance(channel, xr.DataArray):
+            channel_sorted = sorted(channel.data)
```

Comment:
Will the `channel` data array ever be a dask array? If so calling `sorted` on this might be problematic. From my understanding, channel is a dimension all the time, so maybe this is not an issue. A safer way may be calling `channel.values` instead. I think `channel` is small enough to be okay to load into memory.

Reply:
Changed. :)